tests/unit/amo/components/TestInstallButtonWrapper.js

```diff
@@ -836,4 +872,1145 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('Tests for withInstallHelpers', () => {
+    it('calls getAddon() when the component is rendered', () => {
+      const addon = fakeAddon;
+      _loadVersions({ slug: addon.slug, versions: [addon.current_version] });
+
+      const _addonManager = getFakeAddonManagerWrapper();
+
+      render({ _addonManager });
+
+      expect(_addonManager.getAddon).toHaveBeenCalledWith(addon.guid);
+    });
+
+    it('does not call getAddon() if we do not have an addonManager', () => {
+      const addon = fakeAddon;
+      _loadVersions({ slug: addon.slug, versions: [addon.current_version] });
+
+      const _addonManager = getFakeAddonManagerWrapper({
+        hasAddonManager: false,
+      });
+
+      render({ _addonManager });
+
+      expect(_addonManager.getAddon).not.toHaveBeenCalled();
+    });
+
+    it('does not call getAddon() if we do not have an addon', () => {
+      const _addonManager = getFakeAddonManagerWrapper();
+
+      render({ _addonManager, addon: null });
+
+      expect(_addonManager.getAddon).not.toHaveBeenCalled();
+
+      // Basic expectation proving rendering didn't fail
+      expect(screen.getByTagName('div')).toBeInTheDocument();
+
+      // We didn't render a proper install button wrapper though.
+      expect(
+        screen.queryByClassName('InstallButtonWrapper'),
+      ).not.toBeInTheDocument();
+    });
+
+    describe('setCurrentStatus', () => {
+      it('sets the status to ENABLED when an extension is enabled', async () => {
```

Comment:
Some of the `getAddon()` calls are subtly different though. It's possible to also parametrize this, but I am not sure it would make the tests more readable...